### PR TITLE
Add VulnerabilitesByPatchDeadline field to search

### PR DESCRIPTION
**Pull request by ghost**
Add to listings:
- all hosts
- hardware
- virtual servers
- cloud hosts

### CODE REVIEW

**Review by mkurek — APPROVED** [review]

File: src/ralph/admin/filters.py [hunk 1]
```diff
@@ -215,6 +232,21 @@ def choices(self, cl):
         }
 
 
+class VulnerabilitesByPatchDeadline(DateListFilter):
+
+    def queryset(self, request, queryset):
+        from ralph.security.models import SecurityScan
+        if any(self.value()):
+            date_start, date_end = self.to_python(self.value())
```
Comment (mkurek) [inline src/ralph/admin/filters.py]: casting to python dates is not necessary (django will do this)

**Review by mkurek — CHANGES_REQUESTED** [review]

File: src/ralph/admin/filters.py [hunk 2]
```diff
@@ -215,6 +215,25 @@ def choices(self, cl):
         }
 
 
+class VulnerabilitesByPatchDeadline(DateListFilter):
+
+    def queryset(self, request, queryset):
+        from ralph.security.models import SecurityScan
+        if any(self.value()):
+            filter = {}
```
Comment (mkurek) [inline src/ralph/admin/filters.py]: consider naming this variable to sth other than built-in function :)

File: src/ralph/admin/filters.py [hunk 3]
```diff
@@ -215,6 +215,25 @@ def choices(self, cl):
...
+            date_start, date_end = self.value()
+            if date_start:
+                filter['vulnerabilities__patch_deadline__gte'] = date_start
+            if date_end:
+                filter['vulnerabilities__patch_deadline__lte'] = date_end
+            print(filter)
```
Comment (mkurek) [inline src/ralph/admin/filters.py]: rm

**Review by mkurek — CHANGES_REQUESTED** [review]

File: src/ralph/admin/filters.py (hunk 2 above)
Comment (ghost) [reply src/ralph/admin/filters.py]: done

File: src/ralph/admin/filters.py (hunk 3 above)
Comment (ghost) [reply src/ralph/admin/filters.py]: done

**Review by mkurek — APPROVED** [review]

**Review by szok — APPROVED** [review]

**Pull request merged**